### DISCUSSION TITLE
Error: Generator "typegraphql-prisma" failed:  /bin/sh: typegraphql-prisma: command not found

### DISCUSSION THREAD

**Discussion by jakubdonovan** [MichalLytek/typegraphql-prisma#423] [Q&A]
Anyone had any luck gettign typegraphql-prisma to work on 5.5.2?
I installed `pnpm i -D typegraphql-prisma` and put the following generator above my client in the schema
```
generator typegraphql {
  provider = "typegraphql-prisma"
}

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

...and other models
```
then ran 
`pnpm dlx prisma generate` which results in the following error.
> Error: Generator "typegraphql-prisma" failed:
> 
> /bin/sh: typegraphql-prisma: command not found

Yes, generating and using prisma normally without typegraphql-prisma works fine. typegraphql-prisma is present in my node_modules, and downgrading prisma to 5.4.2 didn't work either. i'm on node 20
![Screenshot 2023-11-06 at 13 47 26](https://github.com/MichalLytek/typegraphql-prisma/assets/16073448/259746fa-d7f9-4a95-bfa9-ae5b032ec8b4)

**Comment by Notnobcoder** [comment]
"Installed this locally and it worked for me."

<img width="1632" alt="Screenshot 2024-08-07 at 4 31 08 PM" src="https://github.com/user-attachments/assets/b8ff514a-73f3-4734-8e25-066b7f54e5e2">

**Reply by jbdoster** [reply]
Am I missing something?

The problem is that the library looks to `/bin/sh` from the host.

```
> prisma generate

Error: Generator "typegraphql-prisma" failed:

/bin/sh: typegraphql-prisma: command not found
```

What if I need to onboard new developers to my team? They need to add this library globally too?
What if I need to install in a CI pipeline? We need to manually add this to the runner as well?

Why doesn't this library search for it's shell file in the context of the package environment?

```
> prisma generate

Using shell: node_modules/typegraphql-prisma/.bin/sh/typegraphql-prisma
. . .
```

**Reply by MichalLytek** [reply]
This library is using npm feature of exposing scripts via package.json. So when running prisma command, it can find the generator and run the script.
